### PR TITLE
Fix crash when setting max cores lower than machine cores on Linux

### LINKED ISSUE #27158: ISIS Indirect Data Reduction Crash in S(Q, w) Tab on Ubuntu
### Expected behavior
- Clicking the run button in the s(Q, w) tab should produce an `_sqw` workspace. 
- It should be possible to run through the indirect testing script without issue. 
### Actual behavior
When running through the steps in the Indirect Inelastic testing script, Mantid crashes to desktop when the run button is pressed on the s(Q, w) tab. (Step 21 of part 2)

### Steps to reproduce the behavior
1. Interfaces -> Indirect -> Data Reduction -> ISIS Energy Transfer
2. Enter a valid run number (26184)
3. Click Run
4. S(Q, w) tab
5. Select the workspace just created
6. Click run

### Platforms affected
Only observed on Ubuntu 18.04

**Notes:** Segfault occurs in `SpectrumInfo.cpp` L176

### PR DESCRIPTION
**Pull request by cailafinn**
If the maximum number of cores mantid can use is set to be lower than the number of cores on the machine, certain algorithms would ignore this limit and run extra threads.

As a result, workspaces that were not created with this higher number of threads (created in the main thread) would cause a seg fault when the algorithm attempted to access an index of a vector that was out of range.

The root cause of the issue is that threads created separately from the main one do not keep the value set in the `mantid.user.properties` file for the max number of cores. Since this could cause issues in other areas, fixing this will be delayed until the next release. 

**To test:**

1. Change `mantid.user.properties` so that the max cores setting is less than the number of cores on the machine you are testing on.
1. Interfaces -> Indirect -> Data Reduction -> ISIS Energy Transfer
2. Enter a valid run number (26184)
3. Click Run
4. S(Q, w) tab
5. Select the workspace just created
6. Click run
7. No crash should occur

Fixes #27158

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by DanNixon — COMMENTED** [review]
I'm half tempted to put the additional thread count set into the macros in `MultiThreaded.h`. There could well be other scenarios this is affecting (possibly the issue I was looking at with Q1D).

**Comment by ConorMFinn** [comment]
> I'm half tempted to put the additional thread count set into the macros in `MultiThreaded.h`. There could well be other scenarios this is affecting (possibly the issue I was looking at with Q1D).

The resetting of the thread to the config value? Should that be added as part of this PR?

**Comment by DanNixon** [comment]
> > I'm half tempted to put the additional thread count set into the macros in `MultiThreaded.h`. There could well be other scenarios this is affecting (possibly the issue I was looking at with Q1D).
> 
> The resetting of the thread to the config value? Should that be added as part of this PR?

Yes, that is what I meant to suggest. In hindsight that was not clear.

**Comment by ConorMFinn** [comment]
I keep getting build failures on mac and windows when working with the MultiThreaded macros. I've made an issue to do this as maintenance: #27375

File: Framework/Algorithms/src/SofQWNormalisedPolygon.cpp [hunk 1]
```diff
@@ -398,6 +399,17 @@ void SofQWNormalisedPolygon::exec() {
...
+  /*
+    On linux, the value value that is set at startup for the number of allowed
+    threads when mulitprocessing is not carried over to new execution threads.
+    As a result, mantid can segfault due to accessing workspaces that are
+    created in the main thread, as they have too few indexes that are supposed
+    to match the number of threads.
+
+    Resetting the number of threads here to what is set in the config file is a
+    temporary solution.
+    */
```
Comment (DanNixon) [inline Framework/Algorithms/src/SofQWNormalisedPolygon.cpp]: Could you add a `#TODO` here with a link to the new issue you created.

**Review by DanNixon — CHANGES_REQUESTED** [review]
Happy with the fix given there is little time to much else.

**Review by DanNixon — APPROVED** [review]
:shipit:

**Pull request merged**